src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -201,17 +201,53 @@ public override ClaimsPrincipal ValidateToken(string token, TokenValidationParam
                 throw LogExceptionMessage(new ArgumentException(FormatInvariant(TokenLogMessages.IDX10209, token.Length, MaximumTokenSizeInBytes)));
 
             var samlToken = ValidateSignature(token, validationParameters);
+
+            ValidateToken(token, validationParameters, out validatedToken, samlToken, out ClaimsIdentity identity);
```

Comment:
why are two out params needed?

In general we need the same model for all token handling.